### PR TITLE
Finish GitHub sign in, sign up, connect on client side

### PR DESCRIPTION
**Pull request by INTJT**

### CODE REVIEW

**Comment by INTJT** [comment]
[Finish GitHub sign in, sign up, connect on client side](https://trello.com/c/y87vSZKv/69-finish-github-sign-in-sign-up-connect-on-client-side)

**Comment by INTJT** [comment]
[[Sign-Up page] Sign-up with GitHub](https://trello.com/c/GrGj9maF/47-sign-up-page-sign-up-with-github)

File: client/src/containers/setting-page/social/logic/saga.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import * as actions from './actions';
+import * as userActions from 'containers/user/logic/actions';
+import historyHelper from 'helpers/history.helper';
+
+function* linkToGithub(action: ReturnType<typeof actions.linkToGithub>) {
+	try {
+		const { code } = action;
+		const user: IUser = yield call({ context: githubAuthService, fn: githubAuthService.link }, code);
+		yield put(userActions.setUser({ user }));
+		// yield put(actions.githubError({ error: 'щото пашло нитак' }));
```
Comment (svitlua) [inline client/src/containers/setting-page/social/logic/saga.ts:15]: do you need this line?

File: client/src/containers/sign-in/logic/action-types.ts [hunk 2]
```diff
@@ -1,15 +1,20 @@
+import { IUser } from 'typings/common/IUser';
 import { ISignInForm } from 'typings/sign-in-form';
-import { WebApi } from 'typings/webapi';
 
 export const SIGN_IN_USER = 'SIGN_IN_USER';
-export const SIGN_IN_USER_SUCCESS = 'SIGN_IN_USER_SUCCESS';
-export const SIGN_IN_USER_ERROR = 'SIGN_IN_USER_ERROR';
+export const SIGN_IN_USER_BY_GITHUB = 'SIGN_IN_USER:GITHUB';
+export const SIGN_IN_USER_SUCCESS = 'SIGN_IN_USER:SUCCESS';
+export const SIGN_IN_USER_ERROR = 'SIGN_IN_USER:ERROR';
```
Comment (svitlua) [inline client/src/containers/sign-in/logic/action-types.ts:7]: it would be good to use same style for constants' names: either use "_" or ":" separator

**Review by svitlua — APPROVED** [review]
Overall looks great. Good job!

**Pull request merged**